tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs

```diff
@@ -66,53 +78,107 @@ public void Dispose(bool disposing)
 
             disposed = true;
 
-            rule?.Dispose();
+            wafHandle?.Dispose();
         }
 
-        private static Obj RuleSetFromManifest(List<Obj> argCache)
+        private static Obj CreatObjFromRulesStream(List<Obj> argCache, Stream stream)
         {
-            var assembly = typeof(Waf).Assembly;
-            var resource = assembly.GetManifestResourceStream("Datadog.Trace.AppSec.Waf.rule-set.json");
-            using var reader = new JsonTextReader(new StreamReader(resource));
-            var root = JToken.ReadFrom(reader);
+            using var reader = new StreamReader(stream);
+            using var jsonReader = new JsonTextReader(reader);
+            var root = JToken.ReadFrom(jsonReader);
+
+            if (Log.IsEnabled(LogEventLevel.Debug))
+            {
+                LogRuleDetails(root);
+            }
 
             return Encoder.Encode(root, argCache);
         }
 
-        private static WafHandle NewRule()
+        private static Stream GetRulesManifestStream()
+        {
+            var assembly = typeof(Waf).Assembly;
+            return assembly.GetManifestResourceStream("Datadog.Trace.AppSec.Waf.rule-set.json");
+        }
+
+        private static Stream GetRulesFileStream(string rulesFile)
         {
+            if (!File.Exists(rulesFile))
+            {
+                Log.Error($"AppSec could not find the rules file in path \"{rulesFile}\". AppSec will not run any protections in this application.");
```

Comment:
We should use [structured logging](https://github.com/serilog/serilog/wiki/Structured-Data) instead of embedding the filename into the log message. I think the syntax is something like this. Note it's _not_ using string interpolation, so it won't allocate the new string if error-level logging is disabled. It also allows for parsing the data later.
```suggestion
                Log.Error("AppSec could not find the rules file in path \"{rulesFile}\". AppSec will not run any protections in this application.", rulesFile);
```

Reply:
fixed in 1c0de753d